keps/sig-apps/4026-job-creation-timestamp-annotation/README.md

```diff
@@ -0,0 +1,304 @@
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [X] (R) KEP approvers have approved the KEP status as `implementable`
+- [X] (R) Design details are appropriately documented
+- [X] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [X] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Currently, there is no supported way to get the original/expected initial scheduled timestamp for the job. This KEP proposes to set the the original scheduled time, `CreationTimestamp`, as an annotation in the job metadata.
+
+## Motivation
+
+### Goals
+
+- Set job creation timestamp as an annotation on the job.
+- Adding the annotation should not be disruptive to existing workloads.
+
+### Non-Goals
+
+## Proposal
+
+At a high level, the proposal is to modify the CronJob controller to set the job creation timestamp as a job annotation. The details of this are outlined in the Design Details section below.
+
+Job creation timestamp annotation: `batch.kubernetes.io/job-creation-timestamp`
```

Comment:
I find this confusing.
This feature is mostly about a job created by a CronJob. And `creation-timestamp` conflicts with the .metadata.creationTimestamp field.

Maybe `batch.kubernetes.io/cronjob-scheduled-timestamp`?

Reply:
I agree with the confusion, but also, `scheduled-timestamp` can be confusing. I thought to have it as `batch.kubernetes.io/cronjob-original-scheduled-timestamp` but not sure if this is too long, wdyt?

Reply:
Addressed.